packages/expo/src/hooks/useSSO.ts

```diff
@@ -0,0 +1,106 @@
+import { useSignIn, useSignUp } from '@clerk/clerk-react';
+import type { EnterpriseSSOStrategy, OAuthStrategy, SetActive, SignInResource, SignUpResource } from '@clerk/types';
+import * as AuthSession from 'expo-auth-session';
+import * as WebBrowser from 'expo-web-browser';
+
+import { errorThrower } from '../utils/errors';
+
+export type UseSSOParams = {
+  strategy: OAuthStrategy | EnterpriseSSOStrategy;
+  unsafeMetadata?: SignUpUnsafeMetadata;
+  redirectUrl?: string;
+};
+
+export type StartSSOParams = {
+  identifier?: string;
+  unsafeMetadata?: SignUpUnsafeMetadata;
+  redirectUrl?: string;
+};
+
+export type StartSSOFlowReturnType = {
+  /**
+   * Session ID created upon sign-in completion, or null if incomplete.
+   * If incomplete, use signIn or signUp for next steps like MFA.
+   */
+  createdSessionId: string | null;
+  setActive?: SetActive;
+  signIn?: SignInResource;
+  signUp?: SignUpResource;
+  authSessionResult?: WebBrowser.WebBrowserAuthSessionResult;
+};
+
+export function useSSO(useSSOParams: UseSSOParams) {
+  const { signIn, setActive, isLoaded: isSignInLoaded } = useSignIn();
+  const { signUp, isLoaded: isSignUpLoaded } = useSignUp();
+
+  async function startFlow(startSSOFlowParams: StartSSOParams): Promise<StartSSOFlowReturnType> {
+    if (!isSignInLoaded || !isSignUpLoaded) {
+      return {
+        createdSessionId: null,
+        signIn,
+        signUp,
+        setActive,
+      };
+    }
+
+    let createdSessionId = signIn.createdSessionId;
+
+    const redirectUrl =
```

Comment:
A weird behavior is happening with deep linking back from the web browser tab. On expo web, it flows like this:
- Identity Provider redirects to FAPI /acs
- FAPI /acs redirects to application's redirect URL on that same tab
- Expo closes that tab and redirects back to the app but remains on the same page

On mobile, it's not linking to the actual page on the stack. I'll have to perform more testing tomorrow to better understand that.

Reply:
Update: Deep linking is not working for `useOAuth` as well. On our Expo kickstart repo, the "navigation" upon OAuth callback relies on the `<Redirect />` component on the routing root layout, but if our customers aren't using that, then it's sort of a blocker.